celery_pubsub/pubsub.py

```diff
@@ -84,6 +92,20 @@ def _topic_to_re(topic: str) -> re.Pattern[str]:
 _pubsub_manager: PubSubManager = PubSubManager()
 
 
+def subscribe_to(topic: str) -> typing.Callable:
+    def decorator(
+        func: typing.Callable[..., typing.Any]
+    ) -> typing.Callable[..., typing.Any]:
+        app_name, module_name = func.__module__.split(".", 1)
+        task_name = f"{app_name}.{module_name}.{func.__name__}"
+
+        func = task(name=task_name)(func)
```

Comment:
Here, we might want to only do that if we were given a non-task, normal function. Otherwise, we might change some properties of the task (actually, it seems that Celery supports that, silently, and encapsulates tasks, but I'm not sure what will the end result be).

So, I suggest something like:
```suggestion
        if is_normal_function(func):
            func = task(name=task_name)(func)
```

```python
def is_normal_function(func: Any, /) -> bool:
    """Determine whether a given object is a normal function (as opposed to a task)."""
    return func.__class__ is (lambda _: _).__class__
```

Reply:
use `isinstance(func, Task)` instead of `is_normal_function` ?

Reply:
@WaYdotNET You're right, it seems to work both for when created with the `@` and with class inheritance! (I tested on Celery 5 only)